src/cudadev/DataFormats/SoAStore.h

```diff
@@ -0,0 +1,569 @@
+    _ITERATE_ON_ALL(_DECLARE_SOA_DUMP_INFO, ~, __VA_ARGS__)                                                                         \
+    std::cout << "Final offset = " << offset << " computeDataSize(...): " << computeDataSize(nElements, byteAlignment) << std::endl;\
+    std::cout << std::endl;                                                                                                         \
+  }                                                                                                                                 \
+  /* Helper function used by caller to externally allocate the storage */                                                           \
+  static size_t computeDataSize(size_t nElements, size_t byteAlignment = defaultAlignment) {                                        \
+    size_t ret = 0;                                                                                                                 \
+    _ITERATE_ON_ALL(_ACCUMULATE_SOA_ELEMENT, ~, __VA_ARGS__)                                                                        \
+    return ret;                                                                                                                     \
+  }                                                                                                                                 \
+                                                                                                                                    \
+  /**                                                                                                                               \
+   * Helper/friend class allowing SoA introspection.                                                                                \
+   */                                                                                                                               \
+  struct SoAMetadata {                                                                                                              \
+    friend CLASS;                                                                                                                   \
+    SOA_HOST_DEVICE_INLINE size_t size() const { return parent_.nElements_; }                                                       \
+    SOA_HOST_DEVICE_INLINE size_t byteSize() const { return parent_.byteSize_; }                                                    \
+    SOA_HOST_DEVICE_INLINE size_t byteAlignment() const { return parent_.byteAlignment_; }                                          \
+    SOA_HOST_DEVICE_INLINE std::byte* data() const { return parent_.mem_; }                                                         \
+    SOA_HOST_DEVICE_INLINE std::byte* nextByte() const { return parent_.mem_ + parent_.byteSize_; }                                 \
+    SOA_HOST_DEVICE_INLINE CLASS cloneToNewAddress(std::byte* addr) {                                                               \
```

Comment:
This function looks like it could be `const`
```suggestion
    SOA_HOST_DEVICE_INLINE CLASS cloneToNewAddress(std::byte* addr) const {                                                               \
```

Reply:
Fixed in b50d5a9.